app/main/views/questionnaire.py

```diff
@@ -58,21 +56,22 @@ def post_questionnaire(eq_id, form_type, period_id, collection_id, location):
     if not valid:
         return render_page(location, False)
 
-    next_location = g.questionnaire_manager.get_current_location()
+    navigator = g.questionnaire_manager.navigator
+    next_location = navigator.get_next_location(get_answers(current_user), location)
     metadata = get_metadata(current_user)
     logger.info("Redirecting user to next location %s with tx_id=%s", next_location, metadata["tx_id"])
     return redirect_to_questionnaire_page(eq_id, form_type, period_id, collection_id, next_location)
 
 
-@questionnaire_blueprint.route('previous', methods=['GET'])
+@questionnaire_blueprint.route('summary', methods=["GET"])
 @login_required
-def go_to_previous_page(eq_id, form_type, period_id, collection_id):
-    q_manager = g.questionnaire_manager
-    current_location = q_manager.get_current_location()
-    answers = q_manager.get_answers()
-    previous_location = q_manager.navigator.get_previous_location(answers, current_location)
-    g.questionnaire_manager.go_to(previous_location)
-    return redirect_to_questionnaire_page(eq_id, form_type, period_id, collection_id, previous_location)
+def get_summary(eq_id, form_type, period_id, collection_id):
+    navigator = g.questionnaire_manager.navigator
+    latest_location = navigator.get_latest_location(get_answers(current_user), get_completed_blocks(current_user))
```

Comment:
Do we have a test to check that you can't skip to the summary page if that's what this is supposed to be limiting?